server.go

```diff
@@ -4287,8 +4287,36 @@ func (s *server) fetchLastChanUpdate() func(lnwire.ShortChannelID) (
 
 // applyChannelUpdate applies the channel update to the different sub-systems of
 // the server.
-func (s *server) applyChannelUpdate(update *lnwire.ChannelUpdate) error {
+func (s *server) applyChannelUpdate(update *lnwire.ChannelUpdate, outpoint wire.OutPoint,
+	disabled bool) error {
+
+	// Apply the channel update to the outgoing links in the switch. This
+	// helps keep the switch in sync with certain policy actions like
+	// manual disabling of channel.
+	//
+	// NOTE: disabling a channel only disables outgoing payments, not incoming.
+	var edgePolicy *channeldb.ChannelEdgePolicy
+	_, edgePolicy, edgePolicy2, err := s.graphDB.FetchChannelEdgesByOutpoint(&outpoint)
+	if err != nil {
+		return err
+	}
+	if edgePolicy.Node.PubKeyBytes == s.currentNodeAnn.NodeID {
+		edgePolicy = edgePolicy2
+	}
+
+	s.htlcSwitch.UpdateForwardingPolicies(map[wire.OutPoint]htlcswitch.ForwardingPolicy{
```

Comment:
I don't think you can just take the update message and update it here, it's quite dangerous? We need to first use the auth gossiper to process it and then apply it to our subsystems. iirc, there are a series of checks that ensure the integrity, promptness, etc. of the update message. cc link expert @Crypt-iQ for help.

Reply:
Since this is just grabbing what is already in the graph, it must have already been validated and should be safe as long as the above comments I wrote are addressed.